src/main/java/commands/AddModuleCommand.java

```diff
@@ -0,0 +1,35 @@
+package commands;
+
+import access.Access;
+import manager.admin.Admin;
+import manager.chapter.CardList;
+import manager.module.Module;
+import storage.Storage;
+import ui.Ui;
+
+public class AddModuleCommand extends Command {
+    public static final String COMMAND_WORD = "addmodule";
+    String moduleCode;
+
+    public AddModuleCommand(String moduleCode) {
+        this.moduleCode = moduleCode;
+    }
+
+    @Override
+    public void execute(CardList cards, Ui ui, Access access, Storage storage) {
+        if (access.getModuleLevel() == "") {
+            Admin newAdmin = access.getAdmin();
+            newAdmin.add(new Module(moduleCode));
+            access.setAdmin(newAdmin);
+            storage.createModule(moduleCode);
+        } else {
+            System.out.println("Sorry, you currently are in the module/chapter level, "
+                    + "please go back to Admin level first.");
+        }
```

Comment:
instead of checking if (access.getModuleLevel() == "") first, you can do something like this:
if (!(access.getModuleLevel().equals(""))) {
// print error message
return;
}
// rest of the code that should be executed